packages/legend-data-cube/src/stores/core/__tests__/DataCubeQueryExtendSnapshotBuilder.data-cube-test.ts

```diff
@@ -0,0 +1,109 @@
+/**
+ * Copyright (c) 2020-present, Goldman Sachs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  V1_deserializeValueSpecification,
+  V1_serializeValueSpecification,
+} from '@finos/legend-graph';
+import {
+  ENGINE_TEST_SUPPORT__grammarToJSON_valueSpecification,
+  ENGINE_TEST_SUPPORT__JsonToGrammar_valueSpecification,
+} from '@finos/legend-graph/test';
+import { unitTest } from '@finos/legend-shared/test';
+import { describe, expect, test } from '@jest/globals';
+import { validateAndBuildQuerySnapshot } from '../DataCubeQuerySnapshotBuilder.js';
+import {
+  _col,
+  _deserializeLambda,
+  _function,
+} from '../DataCubeQueryBuilderUtils.js';
+import { DataCubeFunction } from '../DataCubeQueryEngine.js';
+import { Test__DataCubeEngine } from './Test__DataCubeEngine.js';
+import { DataCubeQuery } from '../model/DataCubeQuery.js';
+import { INTERNAL__DataCubeSource } from '../model/DataCubeSource.js';
+import type { OperationSnapshotAnalysisTestCaseWithGrammarIssues } from './DatacubeQuerySnapshotBuilderTestUtils.js';
+
+const cases: OperationSnapshotAnalysisTestCaseWithGrammarIssues[] = [
+  [
+    'simple extend',
+    'extend(~name:c|$c.val->toOne() + 1)',
+    '~name:c|$c.val->toOne() + 1->extend()',
```

Comment:
Wow, these are really bad grammar being produced, even buggy. We should look into how to fix these, else when people save this, the query got serialized to text and we would get into trouble.